app/models/manageiq/providers/openstack/network_manager/cloud_subnet.rb

```diff
@@ -13,46 +21,56 @@ def self.raw_create_subnet(ext_management_system, options)
     raise MiqException::MiqCloudSubnetCreateError, e.to_s, e.backtrace
   end
 
-  def self.validate_create_subnet(ext_management_system)
-    validate_subnet(ext_management_system)
-  end
-
-  def provider_object(connection)
-    connection.subnets.get(ems_ref)
-  end
-
-  def raw_delete_subnet
-    with_provider_object(&:destroy)
-    destroy!
+  def delete_cloud_subnet
+    ext_management_system.with_provider_connection(connection_options(cloud_tenant)) do |service|
+      service.delete_subnet(ems_ref)
+    end
   rescue => e
     _log.error "subnet=[#{name}], error: #{e}"
     raise MiqException::MiqCloudSubnetDeleteError, e.to_s, e.backtrace
   end
 
-  def raw_update_subnet(options)
```

Comment:
Ok, so the what I retained from the discussions is that we do want to keep raw for the method wrapping the raw command itself. I'll change it back. Thanks